### ISSUE TITLE
Losing filter after navigating from result page to account page

### ISSUE THREAD

**Issue by pingshunhuangalex** [Miodec/monkeytype#786]
**Describe the bug**
I haven't found a clear pattern for that, but it happens a lot since 1.6.9.

Basically, if I navigate too fast from the result page to the account page. All my filter would be lost with an error toast showing at the top of the screen. I'm guessing "too fast" means when there's still a spinning wheel for the leaderboard on the result page? Can't pinpoint the root of the cause, but it starts to be quite annoying.

![image](https://user-images.githubusercontent.com/17170199/103470000-77914200-4dc0-11eb-9195-4d31299b324e.png)

**To Reproduce**
1. Do a test
2. See the result page
3. Immediately click on the button at the top of the screen to navigate to the account page
4. See the error

**Expected behavior**
No error and filter should never be lost or reverted

**Comment by Provola23** [comment]
Strange! For me this doesn't happen. I'm using Firefox 84.0.1 on Arch Linux and it's working as expected. I also tried it on Google Chrome 89.0.4356 dev and it's also working as expected.

tl;dr: I can't fix this bug because I can't reproduce it.

**Comment by Miodec** [comment]
Also happens without completing a test and just changing something about the config like mode or mode2.

**Comment by Miodec** [comment]
Also when just going back and forth between settings and account.

**Comment by Miodec** [comment]
Fixed in 23544383af686aaf1e803e09a78df8b9c5b90375
